enhancements/authentication/allowing-uri-scheme-in-oidc-sub-claims.md

```diff
@@ -0,0 +1,167 @@
+---
+title: allowing-uri-scheme-in-oidc-sub-claims
+authors:
+  - "@stlaz"
+reviewers:
+  - "@stts"
+  - "@deads2k"
+approvers:
+  - "@stts"
+  - "@deads2k"
+creation-date: 2020-01-19
+last-updated: 2020-01-19
+status: implementable
+see-also:
+  -
+replaces:
+  -
+superseded-by:
+  -
+---
+
+# Allowing URI Scheme in OIDC `sub` claims
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Summary
+
+Make the oauth-server capable of setting up identities for OIDC identity
+providers that use the URI scheme in the `sub` claim of their ID tokens.
+
+## Motivation
+
+Per the [OIDC specification](https://openid.net/specs/openid-connect-core-1_0-final.html#IDToken)
+of the ID token and its more thorough details described in
+[RFC7519 - JSON Web Token (JWT)](https://tools.ietf.org/html/rfc7519#section-4.1.2),
+the `sub` claim may be represented as both a string or a URI. However, URI by
+definition may contain characters such as `/` and `:` that are hard to digest
+for the `identity.oauth.openshift.io` API, which prevents integration with OIDCs
+that make use of this URI scheme. However, an identity provider that follows all
+specifications should still be usable as such for OpenShift to successfully retrieve
+identities from.
+
+### Goals
+
+1. Make it possible for users of the aforementioned identity providers to log in
+   to OpenShift successfully.
+
+### Non-Goals
+
+None.
+
+## Problem Summary
+
+With OpenShift 4.x, the oauth-server no longer allows [configuring the claim
+of ID tokens](https://github.com/openshift/api/blob/670ac3fc997c2f1d19b8c29ef04f70d6e3d4a59e/osin/v1/types.go#L341)
```

Comment:
to understand: in the 3.x past those customers could configure another claim than `sub` and work around the problem?

Reply:
yes, this is what we are doing currently in our v3 clusters

Reply:
yes, that's how it used to work

Reply:
please clarify in the text